sklearn/metrics/pairwise.py

```diff
@@ -646,19 +647,33 @@ def pairwise_distances_argmin_min(
     """
     X, Y = check_pairwise_arrays(X, Y)
 
-    if metric_kwargs is None:
-        metric_kwargs = {}
-
     if axis == 0:
         X, Y = Y, X
 
-    indices, values = zip(
-        *pairwise_distances_chunked(
-            X, Y, reduce_func=_argmin_min_reduce, metric=metric, **metric_kwargs
+    if metric_kwargs is None:
+        metric_kwargs = {}
+
+    if ArgKmin.is_usable_for(X, Y, metric):
+        values, indices = ArgKmin.get_for(
+            X=X, Y=Y, k=1, metric=metric, metric_kwargs=metric_kwargs
+        ).compute(strategy="auto", return_distance=True)
```

Comment:
I suspect that ArgKmin with k=1 is slower than Argmin since it requires maintaining more complex data structures. We can check that and implement PairwiseDistancesArgmin in a later PR, but maybe add a todo explaining that.

Reply:
In the case of `ArgKMin` with `k=1`, the heaps (containing `(n_test, k)` elements) just behave like arrays. Still, there might be some overhead due to routines calls for heap operations and those calls to be replace with simply updating the (argmin, min) pair of `(n_test, )`-shaped arrays.

I also agree that comparing `ArgKMin(k=1)` with a specialised `ArgMin` is worth trying in another PR.